### ISSUE TITLE
CMake / Makefile tries (and fails) to include DirectX headers on Ubuntu 16.10

### ISSUE THREAD

**Issue by sharazam** [MyGUI/mygui#122]
Hi, I wanted to build MyGUI on Ubuntu 16.10 using GCC 5.4. My configuration file:
```
CMake Version 3.5.1 Page 1 of 1

CMAKE_BACKWARDS_COMPATIBILITY    		2.4                                              
CMAKE_BUILD_TYPE				RelWithDebInfo                                   
CMAKE_INSTALL_PREFIX				/usr/local                                       
EXECUTABLE_OUTPUT_PATH				/home/felix/Downloads/mygui_build                
LIBRARY_OUTPUT_PATH				/home/felix/Downloads/mygui_build                
MYGUI_BUILD_BERKELIUM_PLUGIN			OFF                                              
MYGUI_BUILD_DEMOS				ON                                               
MYGUI_BUILD_DOCS				OFF                                              
MYGUI_BUILD_HIKARI_PLUGIN			OFF                                              
MYGUI_BUILD_PLUGINS				OFF                                              
MYGUI_BUILD_TEST_APP				OFF                                              
MYGUI_BUILD_TOOLS				OFF                                              
MYGUI_BUILD_UNITTESTS				OFF                                              
MYGUI_BUILD_WRAPPER				OFF                                              
MYGUI_CREATE_PACKAGE				OFF                                              
MYGUI_DEPENDENCIES_DIR				Dependencies                                     
MYGUI_DONT_USE_OBSOLETE				OFF                                              
MYGUI_FULL_RPATH				OFF                                              
MYGUI_HIGH_LEVEL_WARNINGS			OFF                                              
MYGUI_INSTALL_DOCS				OFF                                              
MYGUI_INSTALL_MEDIA				OFF                                              
MYGUI_INSTALL_SAMPLES				OFF                                              
MYGUI_INSTALL_SAMPLES_SOURCE			OFF                                              
MYGUI_INSTALL_TOOLS				OFF                                              
MYGUI_RENDERSYSTEM				4                                                
MYGUI_RTTI_DISABLE_TYPE_INFO			OFF                                              
MYGUI_STATIC					ON                                               
MYGUI_USE_FREETYPE				ON                                               
OGRE_BUILD                                                 
OGRE_INCLUDE_DIR                                                 
OGRE_LIBRARIES                                                 
OGRE_LIB_DIR                                                 
OGRE_SOURCE                                                 
SVN_EXECUTABLE					SVN_EXECUTABLE-NOTFOUND                          
```

I made a seperate "build" directory and executed `cmake ..` with the output of:
```
-- Configuring MYGUI 3.2.0
-- Detected g++ 5.4.0

-- Enabling GCC visibility flags
-- Looking for FREETYPE...
-- CMAKE_PREFIX_PATH: Dependencies;/home/felix/Downloads/MyGUI_3.2.0/Build/Dependencies;/home/felix/Downloads/MyGUI_3.2.0/Dependencies;/home/felix/Downloads/MyGUI_3.2.0/Build/../Dependencies;/home/felix/Downloads/MyGUI_3.2.0/../Dependencies;/usr/local
-- Found FREETYPE: /usr/lib/x86_64-linux-gnu/libfreetype.so
-- Looking for OIS...
-- Checking for module 'OIS'
--   No package 'OIS' found
-- Could not locate OIS
-- 
-----------------------------------------------------------------------------
-- The following external packages were located on your system.
-- This installation will have the extra features provided by these packages.
+ freetype
+ Doxygen
-----------------------------------------------------------------------------
-- The following OPTIONAL packages could NOT be located on your system.
-- Consider installing them to enable more features from this software.
+ OIS: Input library needed for the samples <http://sourceforge.net/projects/wgois>
-----------------------------------------------------------------------------

-- Unknown svn revision. It won't be included into build.
-- Configuring done
-- Generating done
-- Build files have been written to: /home/felix/Downloads/MyGUI_3.2.0/Build
```

So I thought, OK, I don't need this OIS thing, I should be ready to build.
The compilation step works, the problem starts happening in the linking step:

```
[ 64%] Linking CXX static library ../lib/libMyGUIEngineStatic.a
[ 64%] Built target MyGUIEngine
[ 65%] Building CXX object Platforms/DirectX11/DirectX11Platform/CMakeFiles/MyGUI.DirectX11Platform.dir/src/MyGUI_DirectX11DataManager.cpp.o
[ 66%] Building CXX object Platforms/DirectX11/DirectX11Platform/CMakeFiles/MyGUI.DirectX11Platform.dir/src/MyGUI_DirectX11RenderManager.cpp.o
/home/felix/Downloads/MyGUI_3.2.0/Platforms/DirectX11/DirectX11Platform/src/MyGUI_DirectX11RenderManager.cpp:6:20: fatal error: d3dx11.h: Datei oder Verzeichnis nicht gefunden
 #include <d3dx11.h>
                    ^
compilation terminated.
```

Why, of course. There is no DirectX on Linux. I sadly do not know how what caused this DirectX header to be included. I specifically set the MYGUI_RENDERSYSTEM to 4 to use OpenGL for rendering.

Any help is appreciated.

**Comment by scrawl** [comment]
> I sadly do not know how what caused this DirectX header to be included. I specifically set the MYGUI_RENDERSYSTEM to 4 to use OpenGL for rendering.

Apparently the meaning of MYGUI_RENDERSYSTEM has changed between versions. Compare:
https://github.com/MyGUI/mygui/blob/MyGUI3.2/Platforms/CMakeLists.txt
https://github.com/MyGUI/mygui/blob/MyGUI3.2.1/Platforms/CMakeLists.txt

In 3.2.0, render system 4 referred to DirectX. In 3.2.1, the meaning was changed (not sure if intentionally), now render system 4 is OpenGL. I assume you're currently using 3.2.0 but referring to the documentation meant for 3.2.1+.

Anyway, you're going to run into another problem; the stable MyGUI versions don't yet support the OpenGL render system on Linux, because the winapi is used to create the renderwindow. In the git version (to be released as 3.2.3 at some point in the future), we use SDL2 on all platforms to create the window, so you can use OpenGL+Linux just fine (see #74).

In short, switch to the git version and all your problems should be fixed.

**Comment by sharazam** [comment]
Thanks, I was able to build now. :)
